perception/autoware_tensorrt_bevdet/src/nvjpegdecoder.cpp

```diff
@@ -0,0 +1,299 @@
+// Copyright 2024 AutoCore, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#ifdef __HAVE_NVJPEG__
+
+#include "autoware/tensorrt_bevdet/nvjpegdecoder.hpp"
+
+#include <chrono>
+
+using std::chrono::duration;
+using std::chrono::high_resolution_clock;
+
+int dev_malloc(void ** p, size_t s)
+{
```

Comment:
The functions `dev_malloc`, `dev_free`, `host_malloc`, and `host_free` are simple wrappers around CUDA functions.  Consider adding error handling and logging within these functions to provide more detailed error information in case of CUDA failures.  This will aid in debugging.